src/hooks/useDittoSingleText.ts

```diff
@@ -1,27 +1,31 @@
 import { useContext } from "react";
-import { DittoContext, SourceDetector } from "../lib/context";
-import { nullError } from "../lib/utils";
+import { Plurals } from "../components/Ditto";
+import { DittoContext, SourceDetector, Variables, Count } from "../lib/context";
+import { nullError, interpolateVariableText } from "../lib/utils";
 
 interface useDittoSingleTextProps {
   projectId?: string;
   textId: string;
+  variables: Variables;
+  count?: Count;
 }
 
 export const useDittoSingleText = (props: useDittoSingleTextProps) => {
-  const { projectId, textId } = props;
+  const { projectId, textId, variables = {}, count } = props;
   const { source, variant, options } = useContext(DittoContext);
 
   if (!projectId) return nullError("No Project ID provided.");
 
   if (variant) {
     const data = source?.[projectId]?.[variant];
     if (data) {
+
       if (SourceDetector.isStructured(data)) {
-        return data[textId].text;
+        return interpolateVariableText(data[textId], variables).text;
       }
 
       if (SourceDetector.isFlat(data)) {
-        return data[textId];
+        return interpolateVariableText(data[textId], variables);
```

Comment:
When the format of `data` is `flat`, which we know it is when `SourceDetector.isFlat(data)` is `true`, then `data[textId]` is actually a string, whereas I believe the logic in `interpolateVariableText` expects something like `{ text: string; variables: Variables; count: Count; }`.

Strongly typing the first argument of `interpolateVariableText` will catch this (as mentioned [here](https://github.com/dittowords/ditto-react/pull/10/files#r784859961))

Reply:
👍